### PR TITLE
Update bank branch data using GHA

### PR DESCRIPTION
**Pull request by aathithya-zepto**
Trello card: https://trello.com/c/Ehbqkn6m/36-dddx-update-bsb-database-with-latest-data-from-the-source

The BSB data updates were manually done by running the rake tasks. However, its better to get this automated.

This PR adds 
1) Github actions script to run the rake task, create PR automatically
2) Rake task to handle `bsb_db.json` and `bsb_bank_list.json` updates by fetching the latest data from [AusPaynet](https://bsb.auspaynet.com.au/)

NOTE: This GHA script does not bumps the version. As, our forked zeptofs/bsb branch is not in sync with the upstream [coinjar/bsb](https://github.com/coinjar/bsb/commits/master/lib/bsb/version.rb).

### CODE REVIEW

File: .github/workflows/update_bank_branch_data.yml
Comment (zubin) [inline .github/workflows/update_bank_branch_data.yml]: Did we find that this had no effect, and that it was `.github/CODEOWNERS` which triggered PR reviewer assignment? If so, consider deleting this line to avoid confusing our future selves. 

```suggestion
```

File: lib/bsb/database_generator.rb
Comment (zubin) [inline lib/bsb/database_generator.rb]: Why was this necessary?

File: lib/tasks/bsb_tasks.rake
Comment (zubin) [inline lib/tasks/bsb_tasks.rake]: Formatting nitpick:

```suggestion
    require "bsb/database_generator"
    require "bsb/bank_list_generator"
```

File: lib/tasks/bsb_tasks.rake
Comment (zubin) [inline lib/tasks/bsb_tasks.rake]: Is there another way to calculate this without depending on ActiveSupport, ie with vanilla ruby?

File: lib/tasks/bsb_tasks.rake
Comment (zubin) [inline lib/tasks/bsb_tasks.rake]: To help readability, consider using ruby's `URI::Parser` for escaping. 

```suggestion
    url = URI::Parser.new.escape("http://bsb.hostedftp.com/~auspaynetftp/BSB/KEY TO ABBREVIATIONS AND BSB NUMBERS (#{filename_month} #{filename_year}).csv")
```

File: .github/workflows/update_bank_branch_data.yml
Comment (aathithya-zepto) [reply .github/workflows/update_bank_branch_data.yml]: CODEOWNERS is to define who is responsible for the code in the repository. I think its giving the necessary access to the team/individual.
However [Code owners doc](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) states, Code owners are automatically requested for review when someone opens a pull request that modifies code that they own. So let me try and test it once by removing `team-reviewers`

File: lib/bsb/database_generator.rb
Comment (aathithya-zepto) [reply lib/bsb/database_generator.rb]: The ftp server's filename has a naming convention which has the `previous month` and a `report number` in it. We only know what the month is and not the report number. This code is to fetch the exact filename with the specified month.

File: .github/workflows/update_bank_branch_data.yml
Comment (zubin) [reply .github/workflows/update_bank_branch_data.yml]: I would be very surprised (and disturbed) if CODEOWNERS content could grant access. 😬

File: lib/bsb/database_generator.rb
Comment (aathithya-zepto) [reply lib/bsb/database_generator.rb]: The output of line 20 would be like this

image

File: .github/workflows/update_bank_branch_data.yml
Comment (aathithya-zepto) [reply .github/workflows/update_bank_branch_data.yml]: Removed `team-reviewers: "@zeptofs/team-send"` from GHA file after testing

File: lib/tasks/bsb_tasks.rake
Comment (aathithya-zepto) [reply lib/tasks/bsb_tasks.rake]: Have used `prev_month` now which doesn't requires ActiveSupport

**Review by zubin — APPROVED** [review]
LGTM 👍 🚀

**Review by zachingle — APPROVED** [review]
LGTM! Once this repo is updated, how will the update get passed into the split repo? Will Renovate/Dependabot pick up the change?

**Comment by zubin** [comment]
> LGTM! Once this repo is updated, how will the update get passed into the split repo? Will Renovate/Dependabot pick up the change?

Unfortunately not, they only seem to update gems sourced from rubygems.org. Gems which point to a repo need to be manually updated. So whenever a PR like this crops up, issue a `bundle update bsb` after it's merged.

**Pull request merged**
